internal/pkg/cli/pipeline_init.go

```diff
@@ -523,24 +420,153 @@ For more information, please refer to: https://git.io/JfDFD.`,
+		if err != nil {
+			return err
 		}
+		stage := manifest.PipelineStage{
+			Name:             env.Name,
+			RequiresApproval: env.Prod,
+		}
+		stages = append(stages, stage)
 	}
-	envs, err := o.store.ListEnvironments(o.appName)
+
+	manifest, err := manifest.NewPipelineManifest(pipelineName, provider, stages)
 	if err != nil {
-		return nil, fmt.Errorf("list environments for application %s: %w", o.appName, err)
+		return fmt.Errorf("generate a pipeline manifest: %w", err)
 	}
-	if len(envs) == 0 {
-		return nil, errNoEnvsInApp
+
+	var manifestExists bool
+	manifestPath, err := o.workspace.WritePipelineManifest(manifest)
+	if err != nil {
+		e, ok := err.(*workspace.ErrFileExists)
+		if !ok {
+			return fmt.Errorf("write pipeline manifest to workspace: %w", err)
+		}
+		manifestExists = true
+		manifestPath = e.FileName
+	}
+
+	manifestPath, err = relPath(manifestPath)
+	if err != nil {
+		return err
+	}
+
+	manifestMsgFmt := "Wrote the pipeline manifest for %s at '%s'\n"
+	if manifestExists {
+		manifestMsgFmt = "Pipeline manifest file for %s already exists at %s, skipping writing it.\n"
+	}
+	log.Successf(manifestMsgFmt, color.HighlightUserInput(o.githubRepo), color.HighlightResource(manifestPath))
+	log.Infoln("The manifest contains configurations for your CodePipeline resources, such as your pipeline stages and build steps.")
+	return nil
+}
+
+func (o *initPipelineOpts) createBuildspec() error {
+	artifactBuckets, err := o.artifactBuckets()
+	if err != nil {
+		return err
+	}
+	content, err := o.parser.Parse(buildspecTemplatePath, struct {
+		BinaryS3BucketPath string
+		Version            string
+		ArtifactBuckets    []artifactBucket
+	}{
+		BinaryS3BucketPath: binaryS3BucketPath,
+		Version:            version.Version,
+		ArtifactBuckets:    artifactBuckets,
+	})
+	if err != nil {
+		return err
+	}
+	buildspecPath, err := o.workspace.WritePipelineBuildspec(content)
+	var buildspecExists bool
+	if err != nil {
+		e, ok := err.(*workspace.ErrFileExists)
+		if !ok {
+			return fmt.Errorf("write buildspec to workspace: %w", err)
+		}
+		buildspecExists = true
+		buildspecPath = e.FileName
+	}
+	buildspecMsgFmt := "Wrote the buildspec for the pipeline's build stage at '%s'\n"
+	if buildspecExists {
+		buildspecMsgFmt = "Buildspec file for pipeline already exists at %s, skipping writing it.\n"
+	}
+	buildspecPath, err = relPath(buildspecPath)
+	if err != nil {
+		return err
 	}
-	return envs, nil
+	log.Successf(buildspecMsgFmt, color.HighlightResource(buildspecPath))
+	log.Infoln("The buildspec contains the commands to build and push your container images to your ECR repositories.")
+
+	return nil
+}
+
+func (o *initPipelineOpts) createSecretName() string {
+	return fmt.Sprintf(fmtSecretName, o.appName, o.githubRepo)
+}
+
+func (o *initPipelineOpts) createPipelineName() string {
+	return fmt.Sprintf(fmtPipelineName, o.appName, o.githubOwner, o.githubRepo)
+}
+
+func (o *initPipelineOpts) createPipelineProvider() (manifest.Provider, error) {
```

Comment:
nit: can we reserve the prefix "create*" if it results in the creation of an entity (like SSM, or a file)? so maybe just `secretName()`, `pipelineName()`, `pipelineProvider()`

Reply:
Changed! And changed the string `secretName` to just `secret`.